## How much water do they Need?

### The Law of the River 

* Upper Basin (7.5 million acre feet per year or 293 $m^3/s$)
    * Colorado 
    * Utah
    * Wyoming 
    * New Mexico 
    * Arizona


* Lower Basin (7.5 million acre feet per year)
    * California
    * Arizona 
    * Nevada 
    
    
    
* Mexico (1.5 maf per year or 58.6 $m^3/s$)

#### Personal Use of Water 

* Personal Water Use is that used in the home, and for purposes sake we are assuming a more luxurious, privilege life for each of our individuals. 

* The resor

In [1]:
import pandas as pd
import numpy as np
import math

#import os
#os.getcwd()

df = pd.read_csv("csvData.csv")
df.head(10)

rank           State       Pop  Growth   Pop2018   Pop2010  \
0     1      California  39613493  0.0038  39461588  37319502   
1     2           Texas  29730311  0.0385  28628666  25241971   
2     3         Florida  21944577  0.0330  21244317  18845537   
3     4        New York  19299981 -0.0118  19530351  19399878   
4     5    Pennsylvania  12804123  0.0003  12800922  12711160   
5     6        Illinois  12569321 -0.0121  12723071  12840503   
6     7            Ohio  11714618  0.0033  11676341  11539336   
7     8         Georgia  10830007  0.0303  10511131   9711881   
8     9  North Carolina  10701022  0.0308  10381615   9574323   
9    10        Michigan   9992427  0.0008   9984072   9877510   

   growthSince2010  Percent   density  
0           0.0615   0.1184  254.2929  
1           0.1778   0.0889  113.8081  
2           0.1644   0.0656  409.2229  
3          -0.0051   0.0577  409.5400  
4           0.0073   0.0383  286.1704  
5          -0.0211   0.0376  226.3967  
6           0.0152   0.0350  286.6944  
7           0.1151   0.0324  188.3054  
8           0.1177   0.0320  220.1041  
9           0.0116   0.0299  176.7351

In [2]:
df2 = df.iloc[[0,13, 19, 29, 31, 36, 51]]
df2

rank       State       Pop  Growth   Pop2018   Pop2010  growthSince2010  \
0      1  California  39613493  0.0038  39461588  37319502           0.0615   
13    14     Arizona   7520103  0.0506   7158024   6407172           0.1737   
19    20    Colorado   5893634  0.0356   5691287   5047349           0.1677   
29    30        Utah   3310774  0.0499   3153550   2775332           0.1929   
31    32      Nevada   3185786  0.0523   3027341   2702405           0.1789   
36    37  New Mexico   2105005  0.0059   2092741   2064552           0.0196   
51    52     Wyoming    581075  0.0060    577601    564487           0.0294   

    Percent   density  
0    0.1184  254.2929  
13   0.0225   66.2016  
19   0.0176   56.8653  
29   0.0099   40.2918  
31   0.0095   29.0195  
36   0.0063   17.3540  
51   0.0017    5.9847

In [3]:
ca_pop = 39613493
az_pop = 7520103
co_pop = 5893634
ut_pop = 3310774
nv_pop = 3185786
nm_pop = 2105005
wy_pop = 581075


total_pop=df2['Pop'].sum()
total_pop

62209870

In [5]:
#Upper Basin gallons per day. Look up nevada, 200 is placeholder
person = 200

ca = 137
co = 163
wy = 217
nv = 197
nm = 142
az = 181
ut = 213
us = 101.75

year = 365 

awu = person*year 

ca_awu = ca*year
co_awu = co*year
wy_awu = wy*year
nm_awu = nm*year
nv_awu = nv*year
az_awu = az*year
ut_awu = ut*year


print("The Average water per person for a year is", awu, "gallons.")

print("The Average water per person for a year is", ca_awu, "gallons.")

print("The Average water per person for a year is", co_awu, "gallons.")

print("The Average water per person for a year is", wy_awu, "gallons.")

print("The Average water per person for a year is", nm_awu, "gallons.")

print("The Average water per person for a year is", az_awu, "gallons.")

print("The Average water per person for a year is", ut_awu, "gallons.")

The Average water per person for a year is 73000 gallons.
The Average water per person for a year is 50005 gallons.
The Average water per person for a year is 59495 gallons.
The Average water per person for a year is 79205 gallons.
The Average water per person for a year is 51830 gallons.
The Average water per person for a year is 66065 gallons.
The Average water per person for a year is 77745 gallons.


In [7]:
pop = awu*total_pop


#Calculating the Water totals per year, for personal Use. 

ca_wt = ca_awu*ca_pop

co_wt = co_awu*co_pop

wy_wt = wy_awu*wy_pop

nm_wt = nm_awu*nm_pop

nv_wt = nv_awu*nv_pop

az_wt = az_awu*az_pop

ut_wt = ut_awu*ut_pop

print('The Average water use for all the states in the Colorado River Compact is', pop, 'gallons.')



print('The Average water use for all Coloradoans', co_wt, 'gallons.')


print('The Average water use for all Wyomians', wy_wt, 'gallons.')


print('The Average water use for all New Mexicans', nm_wt, 'gallons.')


print('The Average water use for Arizonans', az_wt, 'gallons.')


print('The Average water use for all Utahn', ut_wt, 'gallons.')


print('The Average water use for all Californians', ca_wt, 'gallons.')


print('The Average water use for all Nevadans', nv_wt, 'gallons.')

The Average water use for all the states in the Colorado River Compact is 4541320510000 gallons.
The Average water use for all Coloradoans 350641754830 gallons.
The Average water use for all Wyomians 46024045375 gallons.
The Average water use for all New Mexicans 109102409150 gallons.
The Average water use for Arizonans 496815604695 gallons.
The Average water use for all Utahn 257396124630 gallons.
The Average water use for all Californians 1980872717465 gallons.
The Average water use for all Nevadans 229073942330 gallons.


## How much Water is there?

* Approximately, we know there are 325,851 gallons in an acre-foot of water, and following the literature we find the common unit of measure is maf, or million acre feet.
* Since we know the range of gallons people demand for personal use, we can now ask how many acre feet do these same people need. 
* Then how many maf the total population of the Colorado River Compact  need a year. 

In [8]:
maf = 325851*1000000 # one million acre-feet = gallons in one acre-foot times one million.

cm3 = 1233.48*1000000 # one million acre-feet = Cubic meters per second in one acre-foot times one million. 

waf = pop/maf #The amount of water demand for total population of Colorado River Compact in maf.

wcm3 = pop/cm3 #Water demand in cubic meters per second.




ca_waf = ca_wt/maf

co_waf = co_wt/maf

wy_waf = wy_wt/maf

nm_waf = nm_wt/maf

nv_waf = nv_wt/maf

az_waf = az_wt/maf

ut_waf = ut_wt/maf

total_pers_use = ca_waf + co_waf + wy_waf + nm_waf + nv_waf + az_waf + ut_waf 
total_pers_use

#print('The current, total water demand of all the states in the Colorado River \ Compact is around', round(waf,2), 'maf (million acre-feet of water per year).')


#print('The current, total water demand for California (based on 2015 water usage) is around', round(ca_waf,2), 'maf.')

10.648813716928904

In [9]:
print('The current, total water demand of all the states in the Colorado River Compact is around', round(waf,4), 'maf (million acre-feet of water per year). After Calcluations we find the personal use of people dependent on the Colorado River is', round(total_pers_use, 2), 'maf. \n')


print('The current, total water demand for California (based on 2015 Public Supply per capita use, in gallons/person/per day water usage) is around', round(ca_waf,4), 'maf (million acre-feet of water per year). \n')


print('The current, total water demand for Colorado is around', round(co_waf,4), 'maf (million acre-feet of water per year). \n')


print('The current, total water demand for Wyoming is around', round(wy_waf,4), 'maf (million acre-feet of water per year). \n')


print('The current, total water demand for New Mexico is around', round(nm_waf,4), 'maf (million acre-feet of water per year). \n')


print('The current, total water demand for Arizona is around', round(az_waf,4), 'maf (million acre-feet of water per year). \n')


print('The current, total water demand for Utah is around', round(ut_waf,4), 'maf (million acre-feet of water per year). \n')


print('The current, total water demand for Nevada is around', round(nv_waf,4), 'maf (million acre-feet of water per year). \n')


The current, total water demand of all the states in the Colorado River Compact is around 13.9368 maf (million acre-feet of water per year). After Calcluations we find the personal use of people dependent on the Colorado River is 10.65 maf. 

The current, total water demand for California (based on 2015 Public Supply per capita use, in gallons/person/per day water usage) is around 6.0791 maf (million acre-feet of water per year). 

The current, total water demand for Colorado is around 1.0761 maf (million acre-feet of water per year). 

The current, total water demand for Wyoming is around 0.1412 maf (million acre-feet of water per year). 

The current, total water demand for New Mexico is around 0.3348 maf (million acre-feet of water per year). 

The current, total water demand for Arizona is around 1.5247 maf (million acre-feet of water per year). 

The current, total water demand for Utah is around 0.7899 maf (million acre-feet of water per year). 

The current, total water demand f

## Range of Water Usage 

* Reserach shows the water usage for personal usage of water is between 80-200 gallons. 

* 80 gallons per person a day - The current, total water demand of all the states in the Colorado River Compact would be around **5.57 maf (million acre-feet of water per year).** 

* 100 ' -  The current, total water demand of all the states in the Colorado River Compact would be around **6.97 maf (million acre-feet of water per year).** 

* 150 ' - The current, total water demand of all the states in the Colorado River Compact would be around **10.45 maf (million acre-feet of water per year).** 

* 200 ' - The current, total water demand of all the states in the Colorado River Compact would be around **13.94 maf (million acre-feet of water per year).**



For Context the average person uses 101.95 gallons of water a day. Meaning, the most common sized water bottle, around 16.9 ounces, is a part of our collective every conscience. Buying a water from a vendor on the street is an experience most of us are familiar with. Now, the numbers are going to get a bit bigger here, there are 128 fluid ounces in a gallon. 

In [52]:
wb_per_gallon = 128/16.9


type(water_bottles)
int(water_bottles)

ca_wb = ca*int(wb_per_gallon)
co_wb = co*int(wb_per_gallon)
wy_wb = wy*int(wb_per_gallon)
nv_wb = nv*int(wb_per_gallon)
nm_wb = nm*int(wb_per_gallon)
az_wb = az*int(wb_per_gallon)
ut_wb = ut*int(wb_per_gallon)
us_wb = us*int(wb_per_gallon)




print('Average US Water Use in 2015 was 101.95, which would require', round(us_wb,0),'water bottles per day. So if ')
print('Average US Water Use in 2015 for California, which would require', round(ca_wb,0),'water bottles per day')
print('Average US Water Use in 2015 for Colorado, which would require', round(co_wb,0), 'water bottles per day')
print('Average US Water Use in 2015 for Wyoming, which would require', round(wy_wb,0), 'water bottles per day')
print('Average US Water Use in 2015 for Nevada, which would require', round(nv_wb,0), 'water bottles per day')
print('Average US Water Use in 2015 for New Mexico, which would require', round(nm_wb,0), 'water bottles per day')
print('Average US Water Use in 2015 for Arizona, which would require', round(az_wb,0), 'water bottles per day')
print('Average US Water Use in 2015 for Utah, which is would require', round(ut_wb,0),'water bottles per day')


Average US Water Use in 2015 was 101.95, which would require 712.0 water bottles per day
Average US Water Use in 2015 for California, which would require 959 water bottles per day
Average US Water Use in 2015 for Colorado, which would require 1141 water bottles per day
Average US Water Use in 2015 for Wyoming, which would require 1519 water bottles per day
Average US Water Use in 2015 for Nevada, which would require 1379 water bottles per day
Average US Water Use in 2015 for New Mexico, which would require 994 water bottles per day
Average US Water Use in 2015 for Arizona, which would require 1267 water bottles per day
Average US Water Use in 2015 for Utah, which is would require 1491 water bottles per day


#### Millions gallon per day 

* 1 acre-foot = 0.32585 Mgal

* 1 maf = 325850 Mgal


In [12]:
#maf to M/gdal 
mgal_maf = 1000000*0.32585
mgal_maf

325850.0

# State's Water Use 

## Upper Basin 


### Colorado 

In [13]:
co = pd.read_csv("CO.csv") 
co = co.iloc[15:18]
co

state_cd   2s       8     8.1  \
15    Public Supply deliveries to domestic, in Mgal/d  16s  456.22   440.8   
16  Public Supply deliveries to commercial, in Mgal/d  16s  111.84  107.56   
17  Public Supply deliveries to industrial, in Mgal/d  16s   17.74   20.89   

       8.2 8.3     8.4     8.5     8.6  
15  481.28   -  529.59  539.23  636.69  
16  100.73   -       -       -       -  
17   19.42   -       -       -       -

### Utah

In [14]:
ut = pd.read_csv("UT.csv") 
ut = ut.iloc[15:18]
ut

state_cd   2s      49    49.1  \
15    Public Supply deliveries to domestic, in Mgal/d  16s  339.92  359.96   
16  Public Supply deliveries to commercial, in Mgal/d  16s   49.52   61.94   
17  Public Supply deliveries to industrial, in Mgal/d  16s   14.83   21.49   

      49.2 49.3    49.4    49.5    49.6  
15  339.67    -  459.87  453.46  495.77  
16  114.99    -       -       -       -  
17   17.29    -       -       -       -

### Wyoming 

In [15]:
wy = pd.read_csv("wy.csv") 
wy = wy.iloc[15:18]
wy

state_cd   2s     56   56.1  \
15    Public Supply deliveries to domestic, in Mgal/d  16s  61.04   55.2   
16  Public Supply deliveries to commercial, in Mgal/d  16s  13.85  16.37   
17  Public Supply deliveries to industrial, in Mgal/d  16s   3.22   5.46   

     56.2 56.3   56.4   56.5   56.6  
15  54.12    -  71.05  72.73  82.34  
16  16.14    -      -      -      -  
17   2.36    -      -      -      -

### New Mexico 

In [16]:
nm = pd.read_csv("nm.csv")
nm = nm.iloc[15:18]
nm

state_cd   2s      35   35.1  \
15    Public Supply deliveries to domestic, in Mgal/d  16s  179.11  163.4   
16  Public Supply deliveries to commercial, in Mgal/d  16s   41.55  68.35   
17  Public Supply deliveries to industrial, in Mgal/d  16s     2.8  14.63   

      35.2 35.3    35.4    35.5    35.6  
15  188.31    -  175.04  160.21  145.12  
16   77.83    -       -       -       -  
17   14.61    -       -       -       -

## Arizona 

* To proportion allotments. 



In [17]:
az = pd.read_csv("AZ.csv")
az = az.iloc[15:18]
az

state_cd   2s       4     4.1  \
15    Public Supply deliveries to domestic, in Mgal/d  16s  449.11  508.93   
16  Public Supply deliveries to commercial, in Mgal/d  16s   90.39  117.07   
17  Public Supply deliveries to industrial, in Mgal/d  16s   78.73   78.49   

       4.2     4.3     4.4     4.5    4.6  
15  525.85  839.36  887.74  912.01  963.3  
16  134.56       -       -  193.36  231.8  
17   66.27       -       -       0      -

## Lower Basin 


### California 

In [18]:
ca = pd.read_csv("CA.csv")
ca = ca.iloc[15:18]
ca

state_cd   2s        6     6.1  \
15    Public Supply deliveries to domestic, in Mgal/d  16s  3238.47  3740.3   
16  Public Supply deliveries to commercial, in Mgal/d  16s  1221.04   790.7   
17  Public Supply deliveries to industrial, in Mgal/d  16s   493.54  491.99   

        6.2 6.3      6.4      6.5      6.6  
15  3711.23   -  3983.75  3865.64  3227.97  
16   994.12   -        -        -        -  
17   282.52   -        -        -        -

### Nevada

In [19]:
nv = pd.read_csv("nv.csv")
nv = nv.iloc[15:18]
nv

state_cd   2s      32    32.1  \
15    Public Supply deliveries to domestic, in Mgal/d  16s  189.17  238.28   
16  Public Supply deliveries to commercial, in Mgal/d  16s   53.87   89.47   
17  Public Supply deliveries to industrial, in Mgal/d  16s     6.3    2.63   

      32.2 32.3    32.4    32.5    32.6  
15  306.07    -  421.45  330.95  329.72  
16  115.86    -       -       -       -  
17    2.19    -       -       -   10.24

## Jackson Hole

- Teton Country, 2015 used 390 gallons of water per day. 